vm/src/obj/objstr.rs

```diff
@@ -725,25 +726,32 @@ impl PyString {
         !self.value.is_empty() && self.value.chars().all(|c| c.is_ascii_whitespace())
     }
 
+    // Return true if all cased characters in the string are uppercase and there is at least one cased character, false otherwise.
     #[pymethod]
     fn isupper(&self, _vm: &VirtualMachine) -> bool {
-        // TODO: assert not " ".isupper(), assert not "_".isupper()
-        !self.value.is_empty()
-            && self
-                .value
-                .chars()
-                .filter(|x| !x.is_ascii_whitespace() && *x != '_')
-                .all(char::is_uppercase)
+        let mut cased = false;
```

Comment:
how about this way?
```rust
   let mut cased = true;
   for c in self.value.chars() {
       if is_cased(c) && c.is_uppercase() {
            continue
        } else {
           cased = false;
           break;  // or `return true` as original
       }
   }
   cased
```

Reply:
oh it's looks better!

Reply:
this code return `false` when `"111a".islower()`
but in python it will return `true`